lib/graphql/schema/union.rb

```diff
@@ -5,31 +5,52 @@ class Union < GraphQL::Schema::Member
       extend GraphQL::Schema::Member::AcceptsDefinition
 
       class << self
-        def possible_types(*types)
+        def possible_types(*types, context: GraphQL::Query::NullContext, **options)
           if types.any?
-            @possible_types = types
+            types.each do |t|
+              type_memberships << type_membership_class.new(self, t, options)
+            end
           else
-            all_possible_types = @possible_types || []
-            all_possible_types += super if defined?(super)
-            all_possible_types.uniq
+            visible_types = []
+            type_memberships.each do |tv|
+              if tv.visible?(context)
```

Comment:
This should be `tm` for `type_membership`?

Reply:
Derp, yes, failed rename 😅